float32 range
float32 min_angle
float32 max_angle
